astroid/node_classes.py

```diff
@@ -55,10 +56,10 @@
 )
 from astroid.manager import AstroidManager
 
-try:
+if sys.version_info >= (3, 8):
```

Comment:
Should be py38_plus from consts :)

Reply:
`if PY38_PLUS` can't be inferred by `mypy`. I added `not PY310` to the test cases but that's about it what will work.

Reply:
I don't understand, what is the problem with mypy ?

Reply:
`mypy` doesn't really infer if an if-block is reachable. It only checks specific conditions one of them being `sys.version_info`. https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks

Btw we had a similar issue with the overload-guard in `astroid/rebuilder.py`. See #1069

--
For tests `PY310_PLUS` and so on are fine, but for everything else directly using `sys.version_info` would be my preferred way of doing it. Although there is no need to replace all current occurrences, just if / when it makes sense, i.e. to fix some typing issue.

Reply:
We could remove the constant entirely in another MR if they do not make sense (well if they're making mypy worse). Originally I refactored because a single constant is better than n constants with the same name in n files.

Reply:
In most cases they don't change anything. Just here and for the aforementioned overloads in `astroid/rebuilder.py`.
Therefore, I think we have more important things to do than to remove these.

Reply:
We sure do (but those kind of refactor take a single keystroke in pycharm)